lib/wda/utils.js

```diff
@@ -47,43 +48,48 @@ async function resetProjectFile (agentPath, newBundleId) {
   }
 }
 
+const dependenciesUpdateLock = new AsyncLock();
+
 async function checkForDependencies (bootstrapPath, useSsl = false) {
   try {
     let carthagePath = await fs.which('carthage');
     log.debug(`Carthage found: '${carthagePath}'`);
   } catch (err) {
-    log.warn('Carthage not found. Install using `brew install carthage`');
+    log.errorAndThrow('Carthage not found. Install using `brew install carthage`');
   }
-  if (!await fs.hasAccess(`${bootstrapPath}/Carthage`)) {
-    log.debug('Running WebDriverAgent bootstrap script to install dependencies');
-    try {
-      let args = useSsl ? ['-d', '-D'] : ['-d'];
-      await exec('Scripts/bootstrap.sh', args, {cwd: bootstrapPath});
-    } catch (err) {
-      // print out the stdout and stderr reports
-      for (let std of ['stdout', 'stderr']) {
-        for (let line of (err[std] || '').split('\n')) {
-          if (!line.length) {
-            continue;
+  const carthageRoot = `${bootstrapPath}/Carthage`;
+  await dependenciesUpdateLock.acquire(carthageRoot, async () => {
```

Comment:
I tested it. The lock only shows unhandled promise rejection if we don't prefix it with await. Otherwise it's OK and the exception is properly thrown.